release.md

```diff
@@ -1,30 +1,94 @@
+        
 - The **libs release** (ie. _libsinp_ and _libscap_)
 
-The [drivers versioning](#Drivers-versioning) process happens in this repository, but their release process is mainly automated and managed in our [test-infra](https://github.com/falcosecurity/test-infra). Building is implemented there as per the [Driverkit Build Grid](https://github.com/falcosecurity/test-infra/tree/master/driverkit), and drivers are published to https://download.falco.org/?prefix=driver/.
+    The [libs versioning](#libs-versioning) process happens in this repository. This release process is currently limited to the versioning only: and no artifacts are built nor distributed.
+
+Commonly, we plan a release process when needed by [Falco](https://github.com/falcosecurity/falco). In such cases, we usually release both drivers and libs versions simultaneously. Other releases may occur for hotfixes or at the discretion of maintainers. Releases are mainly used to signal points in time where the source code is assumed to be consistent and stable.
+
+Releases are planned using [GitHub milestones](https://github.com/falcosecurity/libs/milestones). The due date indicate when consumers should expected a tagged release.
+
+Completed releases are denoted by a _git tag_ and a corresponding [GitHub release](https://github.com/falcosecurity/libs/releases).
+
+## Release team
+
+The release team consists of a *release manager* and other contributors from the community. Usually, we seek volunteers during our [community calls](https://github.com/falcosecurity/community#community-calls) or in the #falco channel in slack](https://kubernetes.slack.com/messages/falco), and then decide together the release planning.
+
+The release manager's responsibility is to coordinate the release process and track its progress. 
+
+Note that the release manager does not need to be a maintainer. However, two [maintainers](https://github.com/falcosecurity/libs/blob/master/OWNERS) with repository rights must be part of the team to approve PRs, do the git tags, manage GitHub milestones, etc.
+
+## Release Phases
+
+Regardless if it is a driver or a libs release when initiating a new release, we follow a streamlined process inspired by the [Kubernetes Release Phases](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md).
+
+![release-process-overview](docs/img/release-process.svg "Code Freeze to Thaw")
+
+### Preparation
+
+In the preparation phase, all enhancements expected to go into the release should be merged before the code freeze.
+
+When close to the code freeze starting period, maintainers should evaluate which PRs can't be merged in time and eventually postpone them to the subsequent milestone.
```

Comment:
@leogr perhaps adjust to `Approximately 4 weeks prior to the freeze starting period, maintainers evaluate pending features and determine which features shall be integrated into the upcoming release based on a set of criteria`.

Outline criteria as bullet list. What comes to mind could be:

- Amount of user-facing changes
- Non user-facing changes that can affect resource utilization (CPU, memory usage ...)
- Amount of testing required to ensure stability and safety
- Type of feature (nice-to-have refactor vs adds significant new threat detection capabilities or improves stability or performance)
- Important bug fix

Reply:
That's a good point. I'm unsure about the criteria set, but in general, I agree with your suggestion. :+1: 

@falcosecurity/libs-maintainers wdyt?

Reply:
Maybe start with any set of criteria that makes sense now and iterate on it. Just the fact of writing down a set of criteria is a huge win in terms of transparency.

Reply:
done, though still a WIP